pkg/cmd/init/exec.go

```diff
@@ -92,18 +120,31 @@ func (o *Options) run() error {
 	}
 
 	discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(restConfig)
-	out, err = apply.ApplyCustomResouces(dynamicClient, discoveryClient, reader, o.values, o.ClusteradmFlags.DryRun, "", "init/clustermanagers.cr.yaml")
+	out, err = apply.ApplyCustomResouces(dynamicClient, discoveryClient, reader, o.values, o.ClusteradmFlags.DryRun, "", "init/clustermanager.cr.yaml")
 	if err != nil {
 		return err
 	}
 	output = append(output, out...)
 
-	fmt.Printf("please log on spoke and run:\n%s join --hub-token %s.%s --hub-apiserver %s --cluster-name <cluster_name>\n",
+	fmt.Printf("please log on spoke and run:\n%s join --hub-token %s --hub-apiserver %s --cluster-name <cluster_name>\n",
 		helpers.GetExampleHeader(),
-		o.values.Hub.TokenID,
-		o.values.Hub.TokenSecret,
+		token,
 		restConfig.Host,
 	)
 
 	return apply.WriteOutput(o.outputFile, output)
 }
+
+func waitForBootstrapSecret(kubeClient kubernetes.Interface, b wait.Backoff) (secret *corev1.Secret, err error) {
+	err = retry.OnError(b, func(err error) bool {
```

Comment:
i might use watch with fiedselector here. But it is a blocker.

Reply:
I don't know how to use watch as it is not a controller. Do you mean it is a blocker or not?

Reply:
not blocker, sorry. Updated the comment